Mathlib/Topology/LocallyConstant/Algebra.lean

```diff
@@ -417,12 +409,111 @@ def congrLeftRingEquiv [Semiring Z] (e : X ≃ₜ Y) :
 
 /-- `LocallyConstant.congrLeft` as an `AlgEquiv`. -/
 @[simps!]
-noncomputable
 def congrLeftₐ (R : Type*) [CommSemiring R] [Semiring Z] [Algebra R Z] (e : X ≃ₜ Y) :
     LocallyConstant X Z ≃ₐ[R] LocallyConstant Y Z where
   toEquiv := congrLeft e
   __ := comapₐ R ⟨_, e.symm.continuous⟩
 
 end Comap
 
+section Map
+
+variable {Z : Type*}
+
+/-- `LocallyConstant.map` as a `MulHom`. -/
+@[to_additive (attr := simps) "`LocallyConstant.map` as an `AddHom`."]
+def mapMulHom [Mul Y] [Mul Z] (f : Y →ₙ* Z) :
+    LocallyConstant X Y →ₙ* LocallyConstant X Z where
+  toFun := map f
+  map_mul' := by aesop
```

Comment:
I'd suggest avoiding definining unmathematical versions (like `MulHom`) if you aren't going to need them.